### PR TITLE
Collection argument conversion for IProperty

### LINKED ISSUE #262: Add support for filtering by container property
```php
/** @property Type $type {container TypeProperty} */
$collection->findBy(['type' => new Type()]);
```

### LINKED ISSUE #302: Support array of entities as argument for IN operator search
I have code:
```
$pricelist->allPrices->get()->findBy(['tariff' => $tariffs]
```
and getting error:
```
Modifier %any expects value to be pretty much anything, TariffEntity given
```
in Price entity I have definition:
```
@property TariffEntity $tariff {m:1 TariffEntity, oneSided=true}
```
variable `$tariffs` is array of `TariffEntity`. 

I can guess that problem is that `PriceEntity` is m:1 so I can not pass array there. But is there any way how to load multiple items with multiple m:1 relations as one?

### PR DESCRIPTION
**Pull request by hrach**
closes #262 && fixes #302

### CODE REVIEW

File: src/Collection/Helpers/ArrayCollectionHelper.php [hunk 1]
```diff
@@ -11,6 +11,8 @@
 use Closure;
 use DateTimeImmutable;
 use DateTimeInterface;
+use http\Exception\InvalidArgumentException;
```
Comment (mabar) [inline src/Collection/Helpers/ArrayCollectionHelper.php]: Typo? Phpstorm love this class.

File: composer.json [hunk 2]
```diff
@@ -28,6 +28,7 @@
 	"require-dev": {
 		"nette/bootstrap": "~2.3",
 		"nette/tester": "~2.0.0",
+		"marc-mabe/php-enum": "~3.0",
```
Comment (mabar) [inline composer.json]: Love this choice

Comment (hrach) [reply composer.json]: This is only for testing purposes, but iml is now pretty simple :)

File: src/Collection/Helpers/ArrayCollectionHelper.php (hunk 1 above)
Comment (hrach) [reply src/Collection/Helpers/ArrayCollectionHelper.php]: I do PHP not much nowadays, what the hell was this? 😨 Some PhpStorm's bug?

Comment (mabar) [reply src/Collection/Helpers/ArrayCollectionHelper.php]: Idk. In last version it displays this exception always first instead of namespace specific one.

**Pull request merged**